src/ray/raylet/worker_pool.cc

```diff
@@ -827,7 +831,10 @@ Status WorkerPool::RegisterDriver(const std::shared_ptr<WorkerInterface> &driver
   const auto job_id = driver->GetAssignedJobId();
   HandleJobStarted(job_id, job_config);
 
-  if (driver->GetLanguage() == Language::JAVA) {
+  // TODO do we want to support PrestartWorkers in Julia? Might cut down on runtime overhead...
```

Comment:
probably do, but I think it's fine to punt on it for now.

Reply:
any idea what they're for?

Reply:
I think it's so that when you initially create the cluster, it'll start workers before anything has been scheduled (or even a driver process connected) but not 100% sure on that.

Reply:
yeah I just don't know the specific _reason_ for that...I thought it might be something python specific. Maybe we can ask in the ray slack

Reply:
yeah worth a shot.  my _guess_ is that it's some combination of
1. wanting to make the startup experience smoother for new users who might not be used to cluster management / a holdover from when ray was python only
2. the fact that python has a startup/runtime component which makes pre-starting more beneficial than compiled languages like java/cpp

Reply:
we decided in sync not to support prestart workers to de-risk upstreaming this into the ray project later on